assessment/libs/complex.php

```diff
@@ -0,0 +1,636 @@
+<?php
+
+# Library name: complex
+# Functions to handle operations for complex numbers.  Version 1.  October 2020
+# Author:  Amir Tavangar
+#          Humber College, Toronto, ON
+
+# Include loadlibrary("complex") to import the library in your questions.
+
+# The functions in this library can be used to handle operations for complex numbers such as
+    # Converting from standard to polar form and vice versa
+    # finding modulus, argument, and conjugate
+    # Multiplication, division, raising to a power, and taking the nth root
+    # Finding real or complex roots of quadratic equations
+    
+
+global $allowedmacros;
+array_push($allowedmacros,"cx_add","cx_arg", "cx_conj", "cx_div", "cx_format2pol", "cx_format2std", "cx_modul",
+                        "cx_mul", "cx_quadRoot", "cx_prettyquadRoot",  "cx_plot",
+                         "cx_pow", "cx_polEu","cx_pol2std", "cx_root", "cx_std2pol", "cx_sub");
+                         
+
+
+
+//--------------------------------------------Modulus----------------------------------------------------
+// Function: cx_modul(num, [roundto = 12])
+// Returns the modulus (absolute value) of a complex number, which is sqrt(Re^2 + Im^2). 
+//
+// Parameters:
+// num: An array of real and imaginary parts of a complex number given in square brackets: num = array([Re, Im]).
```

Comment:
IF the input is a single complex number, there is no need for a double-layer of arrays here.

`num = [Re, Im]`, which is equivalent to `num = array(Re,Im)`  would be sufficient

And then you could reference the values as `$num[0]` and `$num[1]` rather than requiring double indexing